### PR TITLE
Replacing form extensions by FQCN and deprecate public legacy aliases.

### PR DESCRIPTION
**Pull request by FabienPapet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Factorize dependency injection for form extensions. This also moves some extensions into the correct directory (some were misplaced and in the middle of other form types)
| Type?             | refacto
| Category?         | CO
| BC breaks?        | yes (only for moved files)
| Deprecations?     | yes
| How to test?      | CI 🟢 + tests auto

You can review commit by commit for easier rewiewing,

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]
LGTM :+1:

File: src/PrestaShopBundle/Resources/config/services/bundle/form/form.yml [hunk 1]
```diff
@@ -7,3 +7,16 @@ services:
 
   form.form_builder_modifier:
     class: PrestaShopBundle\Form\FormBuilderModifier
+
```
Comment (mflasquin) [inline src/PrestaShopBundle/Resources/config/services/bundle/form/form.yml]: Maybe you can update this file to make public false by default ?

File: src/PrestaShopBundle/Resources/config/services/core/form/data_formatter.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ services:
   _defaults:
     autowire: true
```
Comment (mflasquin) [inline src/PrestaShopBundle/Resources/config/services/core/form/data_formatter.yml:3]: can you precise default public true or false ?

**Review by mflasquin — DISMISSED** [review]

**Comment by FabienPapet** [comment]
Tests: https://github.com/FabienPapet/ga.tests.ui.pr/actions/runs/4193493824

**Review by matthieu-rolland — APPROVED** [review]

**Review by mflasquin — APPROVED** [review]

**Review by HanaRebaiQA — APPROVED** [review]
Hello @FabienPapet 

The CI are green : OK
The automatic tests are passed : OK

So, it's QA :heavy_check_mark: 

Thank You!

**Comment by nicosomb** [comment]
Thank you @FabienPapet and @HanaRebaiQA !

**Pull request merged**